CommonMark/Parser/BlockMethods.cs

```diff
@@ -625,12 +625,12 @@ public static void IncorporateLine(LineInfo line, ref Block curptr)
                     container = CreateChildBlock(container, line, BlockTag.BlockQuote, first_nonspace);
 
                 }
-                else if (!indented && curChar == '#' && 0 != (matched = Scanner.scan_atx_header_start(ln, first_nonspace, ln.Length, out i)))
+                else if (!indented && curChar == '#' && 0 != (matched = Scanner.scan_atx_heading_start(ln, first_nonspace, ln.Length, out i)))
                 {
 
                     AdvanceOffset(ln, first_nonspace + matched - offset, false, ref offset, ref column);
-                    container = CreateChildBlock(container, line, BlockTag.AtxHeader, first_nonspace);
-                    container.HeaderLevel = i;
+                    container = CreateChildBlock(container, line, BlockTag.AtxHeading, first_nonspace);
+                    container.Heading = new HeadingData(i);
```

Comment:
That was the first thing I tried.

    error CS1612: Cannot modify the return value of 'Block.Heading' because
it is not a variable

On Tue, Dec 29, 2015 at 10:13 PM, Kārlis Gaņģis <notifications@github.com>
wrote:

> In CommonMark/Parser/BlockMethods.cs
> <https://github.com/Knagis/CommonMark.NET/pull/71#discussion_r48566709>:
>
> >                  {
> >
> >                      AdvanceOffset(ln, first_nonspace + matched - offset, false, ref offset, ref column);
> > -                    container = CreateChildBlock(container, line, BlockTag.AtxHeader, first_nonspace);
> > -                    container.HeaderLevel = i;
> > +                    container = CreateChildBlock(container, line, BlockTag.AtxHeading, first_nonspace);
> > +                    container.Heading = new HeadingData(i);
>
> since Heading is a non-null struct, I think it might be better to use
> container.Heading.Level = i; to avoid creating new instance.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Knagis/CommonMark.NET/pull/71/files#r48566709>.
>